solver/src/driver.rs

```diff
@@ -119,13 +119,15 @@ impl Driver {
             let metrics = &self.metrics;
             async move {
                 let start_time = Instant::now();
-                let result =
-                    match tokio::time::timeout_at(auction.deadline.into(), solver.solve(auction))
-                        .await
-                    {
-                        Ok(inner) => inner,
-                        Err(_timeout) => Err(anyhow!("solver timed out")),
-                    };
+                let result = match tokio::time::timeout_at(
+                    auction.deadline.into(),
+                    solver.solve(auction, metrics.clone()),
+                )
+                .await
+                {
+                    Ok(inner) => inner,
+                    Err(_timeout) => Err(anyhow!("solver timed out")),
+                };
```

Comment:
A slightly more succinct version of this would be:
```suggestion
                let result = match tokio::time::timeout_at(
                    auction.deadline.into(),
                    solver.solve(auction, metrics.clone()),
                )
                .await
                .context("solver timed out")?;
```

Reply:
I only noticed because the line got touched :stuck_out_tongue:. Feel free to make a separate PR, as you are right that it is an unrelated change. That being said I'm OK with doing a stealth fix in this PR as well. I'll leave it up to you :smile:.

Reply:
I also get a warning that "result_flattening" is unstable (see issue #70142 <https://github.com/rust-lang/rust/issues/70142> for more information)